http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -150,6 +169,13 @@ public void apply(Request request)
         };
     }
 
+    private static boolean networkIssue(Throwable throwable)
+    {
+        requireNonNull(throwable, "throwable is null");
```

Comment:
I'd remove this, as I think it makes the method less readable. The `getCausalChain()` method will throw NPE. And this is an internal helper method called from `handleIf()` which should never pass a null.